src/optimize.rs

```diff
@@ -712,39 +712,114 @@ mod optimize_tests {
+        });
+    }
+
+    #[bench]
+    fn bench_optimize_example1(bencher: &mut test::Bencher) {
+        let data = b"QR\x83R\x81[\x83h\x81i\x83L\x83\x85\x81[\x83A\x81[\x83\x8b\x83R\x81[\x83h\x81j\
+                     \x82\xc6\x82\xcd\x81A1994\x94N\x82\xc9\x83f\x83\x93\x83\\\x81[\x82\xcc\x8aJ\
+                     \x94\xad\x95\x94\x96\xe5\x81i\x8c\xbb\x8d\xdd\x82\xcd\x95\xaa\x97\xa3\x82\xb5\x83f\
+                     \x83\x93\x83\\\x81[\x83E\x83F\x81[\x83u\x81j\x82\xaa\x8aJ\x94\xad\x82\xb5\x82\xbd\
+                     \x83}\x83g\x83\x8a\x83b\x83N\x83X\x8c^\x93\xf1\x8e\x9f\x8c\xb3\x83R\x81[\x83h\
+                     \x82\xc5\x82\xa0\x82\xe9\x81B\x82\xc8\x82\xa8\x81AQR\x83R\x81[\x83h\x82\xc6\
+                     \x82\xa2\x82\xa4\x96\xbc\x8f\xcc\x81i\x82\xa8\x82\xe6\x82\xd1\x92P\x8c\xea\x81j\
+                     \x82\xcd\x83f\x83\x93\x83\\\x81[\x83E\x83F\x81[\x83u\x82\xcc\x93o\x98^\x8f\xa4\
+                     \x95W\x81i\x91\xe64075066\x8d\x86\x81j\x82\xc5\x82\xa0\x82\xe9\x81BQR\x82\xcd\
+                     Quick Response\x82\xc9\x97R\x97\x88\x82\xb5\x81A\x8d\x82\x91\xac\x93\xc7\x82\xdd\
+                     \x8e\xe6\x82\xe8\x82\xaa\x82\xc5\x82\xab\x82\xe9\x82\xe6\x82\xa4\x82\xc9\x8aJ\
+                     \x94\xad\x82\xb3\x82\xea\x82\xbd\x81B\x93\x96\x8f\x89\x82\xcd\x8e\xa9\x93\xae\
+                     \x8e\xd4\x95\x94\x95i\x8dH\x8f\xea\x82\xe2\x94z\x91\x97\x83Z\x83\x93\x83^\x81[\
+                     \x82\xc8\x82\xc7\x82\xc5\x82\xcc\x8eg\x97p\x82\xf0\x94O\x93\xaa\x82\xc9\x8aJ\
+                     \x94\xad\x82\xb3\x82\xea\x82\xbd\x82\xaa\x81A\x8c\xbb\x8d\xdd\x82\xc5\x82\xcd\x83X\
+                     \x83}\x81[\x83g\x83t\x83H\x83\x93\x82\xcc\x95\x81\x8by\x82\xc8\x82\xc7\x82\xc9\
+                     \x82\xe6\x82\xe8\x93\xfa\x96{\x82\xc9\x8c\xc0\x82\xe7\x82\xb8\x90\xa2\x8aE\x93I\
+                     \x82\xc9\x95\x81\x8by\x82\xb5\x82\xc4\x82\xa2\x82\xe9\x81B";
+        bench_optimize(data, Version::Normal(15), bencher);
+    }
+
+    /// This benchmarks `Parser` and `Optimizer` on base64 data
+    ///
+    /// Benchmark of base64 input seems like a useful data point because:
+    /// * base64 data is a mixture of digits (`Mode::Numeric`), uppercase letters
+    ///   (`Mode::Alphanumeric`), and lowercase letters (`Mode::Byte`) and therefore
+    ///   should provide fair coverage of parsing and optimization code
+    /// * base64 data is somewhat realistic - e.g. one common scenario for QR codes
+    ///   is encoding arbitrary URLs and base64 may appear in query part of URLs.
+    ///
+    /// The specific base64 input data below has been taken from
+    /// https://cryptopals.com/sets/1/challenges/6 and truncated to 2880 characters
+    /// (roughly the upper capacity of `Mode::Byte` segments in version 40, `EcLevel::L`
+    /// according to https://www.qrcode.com/en/about/version.html).
+    #[bench]
+    fn bench_optimize_base64(bencher: &mut test::Bencher) {
+        let data = b"\
```

Comment:
I saw your comment about `include_str` but it should not be an issue with rust `1.48`?
Maybe `include_bytes` instead?
Anyway if it's problematic it's fine to have the data here

Reply:
Thanks for the feedback - putting base64 into the already existing `test_data` folder is a good idea.  Done.